src/factories/interfaces/IRestricted_PIM_Factory_v1.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+// Internal Interfaces
+import {IOrchestrator_v1} from
+    "src/orchestrator/interfaces/IOrchestrator_v1.sol";
+import {IOrchestratorFactory_v1} from
+    "src/factories/interfaces/IOrchestratorFactory_v1.sol";
+import {IBondingCurveBase_v1} from
+    "@fm/bondingCurve/interfaces/IBondingCurveBase_v1.sol";
+
+// Internal Dependencies
+import {ERC20Issuance_v1} from "src/external/token/ERC20Issuance_v1.sol";
+
+/**
+ * @title   Restricted PIM Factory
+ *
+ * @notice  Used to deploy a PIM workflow with a restricted bonding curve with a mechanism to pre-fund
+ *          the required collateral supply and an opinionated initial configuration.
+ *
+ * @dev     More user-friendly way to deploy a PIM workflow with an restricted bonding curve.
+ *          Anyone can pre-fund the required collateral supply for a bonding curve deployment.
+ *          Initial issuance token supply is minted to the deployer.
+ *          The deployer receives the role to interact with the curve.
+ *          Overall control over workflow remains with `initialAdmin` of the role authorizer.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          This contract is experimental in nature and has not been audited.
+ *                          Please use at your own risk!
+ *
+ * @author  Inverter Network
+ */
+interface IRestricted_PIM_Factory_v1 {
+    //--------------------------------------------------------------------------
+    // Errors
+    error InsufficientFunding(uint availableFunding);
+
+    //--------------------------------------------------------------------------
+    // Events
+
+    /// @notice Event emitted when a new PIM workflow is created.
+    /// @param orchestrator The address of the orchestrator.
+    /// @param issuanceToken The address of the issuance token.
+    /// @param deployer The address of the deployer.
+    event PIMWorkflowCreated(
+        address indexed orchestrator,
+        address indexed issuanceToken,
+        address indexed deployer
+    );
+
+    /// @notice Event emitted when new funding is added.
+    /// @param paymaster Address that pays funding.
+    /// @param actor Address that can use new funding.
+    /// @param token Address of token used for funding.
+    /// @param amount Funding amount.
+    event FundingAdded(
+        address indexed paymaster,
```

Comment:
Not sure I like calling it "paymaster" since that term is used in account abstraction. Maybe "sponsor" is better?